st2common/st2common/util/types.py

```diff
@@ -0,0 +1,83 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Based on http://code.activestate.com/recipes/576694/ (MIT license)
```

Comment:
> You can redistribute those libraries as long as you attribute the authors in the README and include the license text files in the source code. If you intend to distribute modified source code of these libraries, then you can't distribute them.

> I would recommend switching to the Apache License and then all you have to do is attribute both developers and include the license/copyright files for each, even if you modify the source. The only difference between the MIT and the Apache License is that people redistributing your program have to attribute you and that it includes sections to prevent people from doing bad stuff with patents.

http://stackoverflow.com/questions/16457299/apache-2-0-and-modified-bsd-libaries-in-mit-project